gitbook-docs/market/economics.md

```diff
@@ -16,109 +18,164 @@ Proof Market constants:
-7. The proof buyer uses takes their proof and uses it in some way.
+1. A proof requester sends a request with a desired price _c\_r_ to the market.
+2. The Proof Market locks _c\_r_ tokens from the buyer's account.
+3. Proof producers send proposals to the market with a price _c\_p <= c\_r_.
+4. The Proof Market matches the request with the proposal of the proof producer.
+5. The proof producer generates a proof and sends it to the market.
+6. The Proof Market verifies proof and pays _c\_r - commission_ tokens to the producer.
+7. The proof requester takes their proof and uses it.
 
 ### Orders status
 
-Bids (orders for buying proofs) and asks (orders for selling proofs) have the following statuses:
-
-* _Created_: order was sent to the marketplace and was not yet matched (i.e. there is no buyer for an ask or proof producer for a bid);
-* _Processing_: order was matched, and Proof Market is waiting while proof will be generated;
-* _Completed_: the proof was generated and verified by Proof Market;
-* _Withdrawn_: order was cancelled before matching or after a) wrong proof was submitted or b) prof was not submitted in the requested time interval.
-
-### Bid/Ask Matching process
-
-Proof Market provides more complex behaviour than one-hop trading. Each deal includes a proof generation stage. This stage may result in the completed deal (proof was provided and verified by Proof Market) or a restart of the matching (if the proof has not been provided or verification has failed). Moreover, some applications require generating proofs in some time interval. Thus, in the future additional parameters will influence the matching algorithm:
-
-* Proof generation time. Proof producers provide time bounds _t\_a_ in which they are able to generate proof for the statement. Users can set the desired proof generation time _t\_b_. Only asks with the _t\_a_ ≤ _t\_b_ can be matched with the user’s bid.
-* Proof producers rating. The rating depends on the rate of the completed orders by the proof producers (i.e. the proof producer has provided correct proof in the requested time interval).
-
-Here are the basic rules of order matching for bid with desired time _t\_b_ and price _p\_b_:
-
-1. Choose asks with _t\_a_ ≤ _t\_b_ and _p\_a_ ≤ _p\_b_.
-2. Sort them by price and proof generation time (priority between these parameters is defined by the user’s bid).
+Requests (orders for buying proofs) and proposals (orders for selling proofs) have the following statuses:
+
+* _Created_: order was sent to the marketplace and was not yet matched,
+  meaning there is no buyer for a proposal or no proof producer for a request.
+* _Processing_: order was matched, and the Proof Market is waiting for the proof generator
+  to provide the proof.
+* _Completed_: the proof was generated and verified by Proof Market.
+* _Withdrawn_: the order was canceled before matching, a wrong proof was submitted, or
+  no proofs were submitted in the requested time interval.
```

Comment:
Say, the proof producer submitted bad proof. The PM tried to verify it and saw that it doesn't verify. Shouldn't it find a different proof?